### PR TITLE
[FSI] FSI application upgrade

### PR DESCRIPTION
**Pull request by rubenzorrilla**
This PR finally upgrades the FSI application to the new standards (model, analysis, ...). First of all my apologies for such a large PR, it became more and more complex while fixing the entire application. Concerning the Kratos core, there are two minor changes silently included in here, one in the QR utility and a fix for the variable redistribution case in MPI.

We also took the change to enhance the robustness of the MVQN schemes by using a cut-off criteria based on a SVD decomposition of one of the observation matrices. It would be nice to do the same for the Trilinos recursive MVQN utility but we left this as a further task.

### CODE REVIEW

**Review by RiccardoRossi — COMMENTED** [review]
it should also work with the old settings no?

**Comment by jginternational** [comment]
GUI branch depends on this /KratosMultiphysics/GiDInterface#480

File: applications/FSIapplication/custom_utilities/mvqn_convergence_accelerator.hpp [hunk 1]
```diff
@@ -373,96 +344,126 @@ class MVQNFullJacobianConvergenceAccelerator: public ConvergenceAccelerator<TSpa
...
-    ///@{
-    ///@}
+    /**
+     * @brief Drop the oldest data columns to append the new data
+     * This function drops the oldest information to append the new data to both observation matrices
+     */
+    void DropAndAppendDataColumns()
+    {
+        // Observation matrices size is equal to the interface DOFs number. Oldest columns are to be dropped.
+        MatrixPointerType p_new_V = Kratos::make_shared<MatrixType>(mProblemSize, mProblemSize);
```
Comment (RiccardoRossi) [inline applications/FSIapplication/custom_utilities/mvqn_convergence_accelerator.hpp:390]: i am accepting but ... here you are copying the entire matrix at every iteration... this is a lot!!

**Review by RiccardoRossi — DISMISSED** [review]

File: applications/FSIapplication/custom_utilities/mvqn_convergence_accelerator.hpp (hunk 1 above)
Comment (rubenzorrilla) [reply applications/FSIapplication/custom_utilities/mvqn_convergence_accelerator.hpp:390]: Is the unique way I found to do that. Indeed I don't think this is a problem, you only `DropAndAppendDataColumns`when your number of iterations is equal to your problem size. This means that the number of DOFs of your FSI interface is equal to the FSI non-linear iterations. Such situation only appears in extremely small toy problems (hopefully...)

**Comment by rubenzorrilla** [comment]
@jcotela BTW, I added a `DummySolveDistributionIteration()` in the redistribution utility. This fixes what we discussed past week.

**Comment by rubenzorrilla** [comment]
@philbucher can you approve this again? (I've just solved the conflicts arising from  #3269)

**Review by philbucher — APPROVED** [review]
sorry for the inconveniences

**Pull request merged**